src/mpi/coll/algorithms/stub/types.h

```diff
@@ -0,0 +1,34 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+typedef MPIR_TSP_dt_t MPIR_ALGO_dt_t;
+
+typedef MPIR_TSP_op_t MPIR_ALGO_op_t;
+
+typedef struct MPIR_ALGO_comm_t {
+    int id; /*unique id for the communicator */
+    MPIR_TSP_comm_t tsp_comm;
+} MPIR_ALGO_comm_t;
```

Comment:
Why is this struct necessary? What is the `id` used for? Can the `contextid` be used instead?

Reply:
Well, this struct is being used to store algo specific information, for example tree information for tree based algorithms. I am not sure if we are using id now, but we were at some point.